app/code/Magento/Catalog/Model/Product.php

```diff
@@ -828,9 +828,6 @@ public function getStoreIds()
         if (!$this->hasStoreIds()) {
             $storeIds = [];
             if ($websiteIds = $this->getWebsiteIds()) {
-                if (!$this->isObjectNew() && $this->_storeManager->isSingleStoreMode()) {
-                    $websiteIds = array_keys($websiteIds);
```

Comment:
Hi @engcom-Charlie, thanks for your PR.

I agree that removing this code solves the issue and, as we read in the issue, this code doesn't seem to have any meaning. Actually, @danielrussob found the point in time where the `array_keys` was introduced causing the error.

But there is a unit test asserting that the method `isSingleStoreMode()` is called once; indeed now it fails because the method is no more called.

So in order for this PR to be processed, we also need to remove that expectation from the failing unit test.

Reply:
I confirm